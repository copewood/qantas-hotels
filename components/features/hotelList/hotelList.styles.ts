import styled from 'styled-components';

export const ListContainer = styled.div`
  padding: 5rem 0;
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;