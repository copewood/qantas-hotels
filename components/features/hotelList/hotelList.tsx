import React, { FC } from 'react';

import { HotelCard } from '../../../components/features/hotelCard/hotelCard';
import { StyledListContainer } from './hotelList.styles';
import { Hotel } from '../../models/hotel';

type HotelListProps = {
   hotels: Array<Hotel>
   sortOrder: string;
};

export const HotelList: FC<HotelListProps> = ({hotels, sortOrder}) => {
  // TODO sort the list
  return <>
    <StyledListContainer>
      {hotels.map((hotel: Hotel, index: number) => {
        return <HotelCard key={`hotel-${index}`} hotel={hotel}/>
      })}
    </StyledListContainer>
  </>;
};
