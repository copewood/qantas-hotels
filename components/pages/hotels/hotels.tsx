import React, { FC, useEffect, useState} from 'react';

import { HotelListHeader } from 'components/features/hotelListHeader/hotelListHeader';
import { HotelList } from 'components/features/hotelList/hotelList';
import { Hotel } from '@components/models/hotel';

type HotelsProps = {};

const fetchHotels = async () => {
    return fetch('/api/hotels')
        .then((response) => response.json())
        .then((responseJson) => responseJson);
}

export const Hotels: FC<HotelsProps> = ({}) => {
  const [hotels, setHotels] = useState<Array<Hotel>>();

  useEffect(() => {
    fetchHotels().then((response)=>{
      setHotels(response.results as Array<Hotel>)
    });
  },[])
  
  return <>
    {hotels && <>
      <HotelListHeader hotels={hotels}/>
      {/* <HotelList hotels={hotels}/> */}
    </>}
  </>;
};
