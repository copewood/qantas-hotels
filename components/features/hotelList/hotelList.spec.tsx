import React from 'react';
import { render } from '@testing-library/react';
import { HotelList } from './hotelList';
import { mockHotels } from '../../../pages/api/hotels';

describe('HotelList', () => {
  it('should show HotelList', () => {
    const { container } = render(<HotelList hotels={mockHotels.results}/>);
    expect(container).toMatchSnapshot()
  });
});
