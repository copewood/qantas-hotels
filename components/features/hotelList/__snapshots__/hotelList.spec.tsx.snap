// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelList should show HotelList 1`] = `
<div>
  <div
    class="sc-crXcEl jAlXKY"
  >
    <div
      class="sc-eCYdqJ tFDdR"
    >
      <div
        class="sc-bczRLJ bxsSWL"
      >
        <img
          class="sc-gsnTZi bFjRRX"
          src="https://unsplash.it/145/125/?random"
        />
        <div
          class="sc-dkzDqf iuODgQ"
        >
          Exclusive Deal
        </div>
      </div>
      <div
        class="sc-jSMfEi jAQJee"
      >
        <div
          class="sc-iBkjds dqsmth"
        >
          <span>
            Courtyard by Marriott Sydney-North Ryde
          </span>
          <span
            class="sc-hKMtZM gBiuEH"
          >
            Star rating 4.5
          </span>
        </div>
        <div
          class="sc-ftvSup hGwNMW"
        >
          <span>
            7-11 Talavera Rd
          </span>
          <span>
            North Ryde
          </span>
        </div>
        <a
          class="sc-papXJ hierex"
          href="/"
        >
          Deluxe Balcony Room
        </a>
        <div
          class="sc-jqUVSM cERLEw"
        />
      </div>
      <div
        class="sc-gKXOVf eDpLKX"
      >
        <div>
          1 night total (AUD)
        </div>
        <div
          class="sc-kDDrLX DgOrb"
        >
          $
          329
        </div>
        <div
          class="sc-iqcoie esPlUa"
        >
          Save $
          30
        </div>
      </div>
    </div>
    <div
      class="sc-eCYdqJ tFDdR"
    >
      <div
        class="sc-bczRLJ bxsSWL"
      >
        <img
          class="sc-gsnTZi bFjRRX"
          src="https://unsplash.it/145/125/?random"
        />
        <div
          class="sc-dkzDqf iuODgQ"
        >
          Exclusive Deal
        </div>
      </div>
      <div
        class="sc-jSMfEi jAQJee"
      >
        <div
          class="sc-iBkjds dqsmth"
        >
          <span>
            Primus Hotel Sydney
          </span>
          <span
            class="sc-hKMtZM gBiuEH"
          >
            Star rating 5
          </span>
        </div>
        <div
          class="sc-ftvSup hGwNMW"
        >
          <span>
            339 Pitt St
          </span>
          <span>
            Sydney
          </span>
        </div>
        <a
          class="sc-papXJ hierex"
          href="/"
        >
          Deluxe King
        </a>
        <div
          class="sc-jqUVSM cERLEw"
        >
          <span>
            Free Cancellation
          </span>
        </div>
      </div>
      <div
        class="sc-gKXOVf eDpLKX"
      >
        <div>
          1 night total (AUD)
        </div>
        <div
          class="sc-kDDrLX DgOrb"
        >
          $
          375
        </div>
        <div
          class="sc-iqcoie esPlUa"
        >
          Save $
          28
        </div>
      </div>
    </div>
    <div
      class="sc-eCYdqJ tFDdR"
    >
      <div
        class="sc-bczRLJ bxsSWL"
      >
        <img
          class="sc-gsnTZi bFjRRX"
          src="https://unsplash.it/145/125/?random"
        />
        <div
          class="sc-dkzDqf iuODgQ"
        >
          Red Hot
        </div>
      </div>
      <div
        class="sc-jSMfEi jAQJee"
      >
        <div
          class="sc-iBkjds dqsmth"
        >
          <span>
            Rydges World Square Sydney
          </span>
          <span
            class="sc-hKMtZM gBiuEH"
          >
            Star rating 4
          </span>
        </div>
        <div
          class="sc-ftvSup hGwNMW"
        >
          <span>
            389 Pitt St
          </span>
          <span>
            Sydney
          </span>
        </div>
        <a
          class="sc-papXJ hierex"
          href="/"
        >
          Deluxe King Room
        </a>
        <div
          class="sc-jqUVSM cERLEw"
        />
      </div>
      <div
        class="sc-gKXOVf eDpLKX"
      >
        <div>
          1 night total (AUD)
        </div>
        <div
          class="sc-kDDrLX DgOrb"
        >
          $
          227
        </div>
      </div>
    </div>
    <div
      class="sc-eCYdqJ tFDdR"
    >
      <div
        class="sc-bczRLJ bxsSWL"
      >
        <img
          class="sc-gsnTZi bFjRRX"
          src="https://unsplash.it/145/125/?random"
        />
        <div
          class="sc-dkzDqf iuODgQ"
        >
          Exclusive Deal
        </div>
      </div>
      <div
        class="sc-jSMfEi jAQJee"
      >
        <div
          class="sc-iBkjds dqsmth"
        >
          <span>
            PARKROYAL Darling Harbour Sydney
          </span>
          <span
            class="sc-hKMtZM gBiuEH"
          >
            Star rating 4.5
          </span>
        </div>
        <div
          class="sc-ftvSup hGwNMW"
        >
          <span>
            150 Day Street
          </span>
          <span>
            Sydney
          </span>
        </div>
        <a
          class="sc-papXJ hierex"
          href="/"
        >
          Deluxe King
        </a>
        <div
          class="sc-jqUVSM cERLEw"
        >
          <span>
            Free Cancellation
          </span>
        </div>
      </div>
      <div
        class="sc-gKXOVf eDpLKX"
      >
        <div>
          1 night total (AUD)
        </div>
        <div
          class="sc-kDDrLX DgOrb"
        >
          $
          535
        </div>
      </div>
    </div>
    <div
      class="sc-eCYdqJ tFDdR"
    >
      <div
        class="sc-bczRLJ bxsSWL"
      >
        <img
          class="sc-gsnTZi bFjRRX"
          src="https://unsplash.it/145/125/?random"
        />
        <div
          class="sc-dkzDqf iuODgQ"
        >
          Bonus Points
        </div>
      </div>
      <div
        class="sc-jSMfEi jAQJee"
      >
        <div
          class="sc-iBkjds dqsmth"
        >
          <span>
            Metro Hotel Marlow Sydney Central
          </span>
          <span
            class="sc-hKMtZM gBiuEH"
          >
            Star rating 3.5
          </span>
        </div>
        <div
          class="sc-ftvSup hGwNMW"
        >
          <span>
            431-439 Pitt Street
          </span>
          <span>
            Sydney
          </span>
        </div>
        <a
          class="sc-papXJ hierex"
          href="/"
        >
          Deluxe King
        </a>
        <div
          class="sc-jqUVSM cERLEw"
        >
          <span>
            Free Cancellation
          </span>
        </div>
      </div>
      <div
        class="sc-gKXOVf eDpLKX"
      >
        <div>
          1 night total (AUD)
        </div>
        <div
          class="sc-kDDrLX DgOrb"
        >
          $
          295
        </div>
      </div>
    </div>
  </div>
</div>
`;
